language-plutus-core/src/Language/PlutusCore/Constant/Apply.hs

```diff
@@ -78,10 +77,10 @@ instance Monad ConstAppResult where
     ConstAppStuck     >>= _ = ConstAppStuck
     ConstAppError err >>= _ = ConstAppError err
 
-type ConstAppResultDef = ConstAppResult (Value TyName Name ())
+type ConstAppResultDef = ConstAppResult (Term TyName Name ())
```

Comment:
`type` synonyms bitrot too quickly... Perhaps it would be better to turn `Value` into a `newtype` wrapper.

Reply:
How do you mean?

Reply:
Currently we have `type Value = Term` and we use `Value` to indicate that a term is evaluated up to WHNF, but things change constantly and something that is no longer supposed to be a value is still marked as such. A `newtype` would help here.